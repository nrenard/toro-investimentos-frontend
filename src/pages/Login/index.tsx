import React, { FormEvent, useEffect, useState } from 'react';

import Input from 'components/Input';
import Button from 'components/Button';
import Loader from 'components/Loader';

import logo from 'assets/images/logo.png';

import { useUser } from 'containers/user';

import {
  ModalLogin,
  WrapperContent,
  WrapperForm,
  ErrorMessage,
} from './styles';

const Login: React.FC = () => {
  const {
    actions: { makeLogin, makeRegister, resetError },
    data: { loading, error },
  } = useUser();

  const [name, setName] = useState('');
  const [mail, setMail] = useState('');
  const [password, setPassword] = useState('');
  const [isLogin, setIsLogin] = useState(true);

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (isLogin) {
      makeLogin({ mail, password });
    } else {
      makeRegister({ name, mail, password });
    }
  };

  useEffect(() => {
    setName('');
    setMail('');
    setPassword('');
    resetError();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isLogin]);

  return (
    <ModalLogin>
      <WrapperContent>
        <img src={logo} alt="logo" />

        <h1>{isLogin ? 'Acesse sua conta Toro:' : 'Cadastre-se'}</h1>

        {loading ? (
          <Loader />
        ) : (
          <WrapperForm onSubmit={handleSubmit}>
            {!isLogin && (
              <Input
                value={name}
                type="string"
                onChange={setName}
                placeholder="Nome"
                required
                name="name"
              />
            )}

            <Input
              value={mail}
              type="email"
              onChange={setMail}
              placeholder="E-mail"
              required
              name="mail"
            />

            <Input
              value={password}
              onChange={setPassword}
              placeholder="Senha"
              type="password"
              required
              minLength={6}
              maxLength={20}
              name="password"
            />

            {error && <ErrorMessage>{error}</ErrorMessage>}

            {isLogin ? (
              <p onClick={() => setIsLogin(false)}>
                Não é cadastrado? Faça seu cadastro <strong>aqui.</strong>
              </p>
            ) : (
              <p onClick={() => setIsLogin(true)}>
                Já é cadastrado? Faça seu login <strong>aqui.</strong>
              </p>
            )}

            <Button type="submit" width="300px">
              {isLogin ? 'Entrar' : 'Continue'}
            </Button>
          </WrapperForm>
        )}
      </WrapperContent>
    </ModalLogin>
  );
};

export default Login;
