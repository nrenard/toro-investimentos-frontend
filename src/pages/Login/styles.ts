import styled from 'styled-components';

import { Container } from 'components/Input/styles';

import { ButtonStyles } from 'components/Button/styles';

import { simpleFlex } from 'styles/mixins';

export const ModalLogin = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  padding: 0;
  width: 100%;
  height: 100%;
  ${simpleFlex};
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.9);

  @media (min-width: 768px) {
    padding: 50px;
  }
`;

export const WrapperContent = styled.div`
  background-color: ${({ theme }) => theme?.colors?.secondary};
  width: 100%;
  height: 100%;
  border-radius: ${({ theme }) => theme?.base?.radius * 2}px;
  ${simpleFlex};
  flex-direction: column;
  padding: 0 10px;

  img {
    margin-top: ${({ theme }) => theme?.base?.margin * 6}px;
    margin-bottom: ${({ theme }) => theme?.base?.margin * 8}px;
  }

  h1 {
    font-size: 20px;
  }

  @media (min-width: 768px) {
    width: 70%;

    h1 {
      font-size: 28px;
    }
  }
`;

export const WrapperForm = styled.form`
  ${simpleFlex};
  width: 100%;
  justify-content: center;
  flex-direction: column;

  margin-top: ${({ theme }) => theme?.base?.margin * 3}px;

  ${Container} {
    + ${Container} {
      margin-top: ${({ theme }) => theme?.base?.margin}px;
    }
  }

  ${ButtonStyles} {
    margin-top: ${({ theme }) => theme?.base?.margin * 2}px;
  }

  p {
    margin-top: ${({ theme }) => theme?.base?.margin * 2}px;
    cursor: pointer;
  }

  @media (min-width: 768px) {
    width: 70%;
  }
`;

export const ErrorMessage = styled.strong`
  margin-top: ${({ theme }) => theme?.base?.margin * 2}px;
  color: ${({ theme }) => theme?.colors?.error};
`;
