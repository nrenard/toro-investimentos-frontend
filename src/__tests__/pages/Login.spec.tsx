import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import withUser, { initialState } from 'containers/user';

import Login from 'pages/Login';

const Container = withUser(Login);

describe('Login page', () => {
  it('should be render step Login', () => {
    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('should be render step Login to Register', () => {
    const result = render(<Container />);

    const { container } = result;

    const buttonToggleStep: any = container.querySelector('p');

    fireEvent.click(buttonToggleStep);

    expect(result).toMatchSnapshot();
  });

  it('should be render step login to Register to Login', () => {
    const result = render(<Container />);

    const { container } = result;

    const buttonToggleStep: any = container.querySelector('p');

    fireEvent.click(buttonToggleStep);

    fireEvent.click(buttonToggleStep);

    expect(result).toMatchSnapshot();
  });

  it('should be render with error', () => {
    const ContainerWithError = withUser(Login, {
      ...initialState,
      error: 'true',
    });

    const result = render(<ContainerWithError />);

    expect(result).toMatchSnapshot();
  });

  it('should be render with loading', () => {
    const ContainerWithError = withUser(Login, {
      ...initialState,
      loading: true,
    });

    const result = render(<ContainerWithError />);

    expect(result).toMatchSnapshot();
  });

  it('should be submit form in step Login', () => {
    const result = render(<Container />);

    const { container } = result;

    const form: any = container.querySelector('form');
    const mailInput: any = container.querySelector('input[name="mail"]');
    const passInput: any = container.querySelector('input[name="password"]');

    fireEvent.change(mailInput, { target: { value: 'a@a.com' } });
    fireEvent.change(passInput, { target: { value: 'teste123456' } });

    fireEvent.submit(form);

    expect(result).toMatchSnapshot();
  });

  it('should be submit form in step Register', () => {
    const result = render(<Container />);

    const { container } = result;

    const buttonToggleStep: any = container.querySelector('p');

    fireEvent.click(buttonToggleStep);

    const form: any = container.querySelector('form');
    const nameInput: any = container.querySelector('input[name="name"]');
    const mailInput: any = container.querySelector('input[name="mail"]');
    const passInput: any = container.querySelector('input[name="password"]');

    fireEvent.change(nameInput, { target: { value: 'Teste teste' } });
    fireEvent.change(mailInput, { target: { value: 'a@a.com' } });
    fireEvent.change(passInput, { target: { value: 'teste123456' } });

    fireEvent.submit(form);

    expect(result).toMatchSnapshot();
  });
});
