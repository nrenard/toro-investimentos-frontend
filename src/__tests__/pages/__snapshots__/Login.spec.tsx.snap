// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should be render step Login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Acesse sua conta Toro:
          </h1>
          <form
            class="sc-iCoGMd dNxoh"
          >
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                name="mail"
                placeholder="E-mail"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                maxlength="20"
                minlength="6"
                name="password"
                placeholder="Senha"
                required=""
                type="password"
                value=""
              />
            </div>
            <p>
              Não é cadastrado? Faça seu cadastro 
              <strong>
                aqui.
              </strong>
            </p>
            <button
              class="sc-dlnjwi xZsCq"
              type="submit"
              width="300px"
            >
              Entrar
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Acesse sua conta Toro:
        </h1>
        <form
          class="sc-iCoGMd dNxoh"
        >
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              name="mail"
              placeholder="E-mail"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              maxlength="20"
              minlength="6"
              name="password"
              placeholder="Senha"
              required=""
              type="password"
              value=""
            />
          </div>
          <p>
            Não é cadastrado? Faça seu cadastro 
            <strong>
              aqui.
            </strong>
          </p>
          <button
            class="sc-dlnjwi xZsCq"
            type="submit"
            width="300px"
          >
            Entrar
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login page should be render step Login to Register 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Cadastre-se
          </h1>
          <form
            class="sc-iCoGMd dNxoh"
          >
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                name="name"
                placeholder="Nome"
                required=""
                type="string"
                value=""
              />
            </div>
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                name="mail"
                placeholder="E-mail"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                maxlength="20"
                minlength="6"
                name="password"
                placeholder="Senha"
                required=""
                type="password"
                value=""
              />
            </div>
            <p>
              Já é cadastrado? Faça seu login 
              <strong>
                aqui.
              </strong>
            </p>
            <button
              class="sc-dlnjwi xZsCq"
              type="submit"
              width="300px"
            >
              Continue
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Cadastre-se
        </h1>
        <form
          class="sc-iCoGMd dNxoh"
        >
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              name="name"
              placeholder="Nome"
              required=""
              type="string"
              value=""
            />
          </div>
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              name="mail"
              placeholder="E-mail"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              maxlength="20"
              minlength="6"
              name="password"
              placeholder="Senha"
              required=""
              type="password"
              value=""
            />
          </div>
          <p>
            Já é cadastrado? Faça seu login 
            <strong>
              aqui.
            </strong>
          </p>
          <button
            class="sc-dlnjwi xZsCq"
            type="submit"
            width="300px"
          >
            Continue
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login page should be render step login to Register to Login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Acesse sua conta Toro:
          </h1>
          <form
            class="sc-iCoGMd dNxoh"
          >
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                name="mail"
                placeholder="E-mail"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                maxlength="20"
                minlength="6"
                name="password"
                placeholder="Senha"
                required=""
                type="password"
                value=""
              />
            </div>
            <p>
              Não é cadastrado? Faça seu cadastro 
              <strong>
                aqui.
              </strong>
            </p>
            <button
              class="sc-dlnjwi xZsCq"
              type="submit"
              width="300px"
            >
              Entrar
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Acesse sua conta Toro:
        </h1>
        <form
          class="sc-iCoGMd dNxoh"
        >
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              name="mail"
              placeholder="E-mail"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              maxlength="20"
              minlength="6"
              name="password"
              placeholder="Senha"
              required=""
              type="password"
              value=""
            />
          </div>
          <p>
            Não é cadastrado? Faça seu cadastro 
            <strong>
              aqui.
            </strong>
          </p>
          <button
            class="sc-dlnjwi xZsCq"
            type="submit"
            width="300px"
          >
            Entrar
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login page should be render with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Acesse sua conta Toro:
          </h1>
          <form
            class="sc-iCoGMd dNxoh"
          >
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                name="mail"
                placeholder="E-mail"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="sc-bdnxRM eaWVJw"
            >
              <input
                class="sc-gtsrHT kyisFr"
                maxlength="20"
                minlength="6"
                name="password"
                placeholder="Senha"
                required=""
                type="password"
                value=""
              />
            </div>
            <p>
              Não é cadastrado? Faça seu cadastro 
              <strong>
                aqui.
              </strong>
            </p>
            <button
              class="sc-dlnjwi xZsCq"
              type="submit"
              width="300px"
            >
              Entrar
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Acesse sua conta Toro:
        </h1>
        <form
          class="sc-iCoGMd dNxoh"
        >
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              name="mail"
              placeholder="E-mail"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="sc-bdnxRM eaWVJw"
          >
            <input
              class="sc-gtsrHT kyisFr"
              maxlength="20"
              minlength="6"
              name="password"
              placeholder="Senha"
              required=""
              type="password"
              value=""
            />
          </div>
          <p>
            Não é cadastrado? Faça seu cadastro 
            <strong>
              aqui.
            </strong>
          </p>
          <button
            class="sc-dlnjwi xZsCq"
            type="submit"
            width="300px"
          >
            Entrar
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login page should be render with loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Acesse sua conta Toro:
          </h1>
          <div
            class="sc-hKFxyN iKaQIC"
          >
            <div
              class="sc-eCApnc dCsrWr"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Acesse sua conta Toro:
        </h1>
        <div
          class="sc-hKFxyN iKaQIC"
        >
          <div
            class="sc-eCApnc dCsrWr"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login page should be submit form in step Login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Acesse sua conta Toro:
          </h1>
          <div
            class="sc-hKFxyN iKaQIC"
          >
            <div
              class="sc-eCApnc dCsrWr"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Acesse sua conta Toro:
        </h1>
        <div
          class="sc-hKFxyN iKaQIC"
        >
          <div
            class="sc-eCApnc dCsrWr"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login page should be submit form in step Register 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSFjdj jvsuTt"
      >
        <div
          class="sc-gKAaRy ftNFfT"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Cadastre-se
          </h1>
          <div
            class="sc-hKFxyN iKaQIC"
          >
            <div
              class="sc-eCApnc dCsrWr"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSFjdj jvsuTt"
    >
      <div
        class="sc-gKAaRy ftNFfT"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Cadastre-se
        </h1>
        <div
          class="sc-hKFxyN iKaQIC"
        >
          <div
            class="sc-eCApnc dCsrWr"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
